include/openthread/link_raw.h

```diff
@@ -267,7 +267,7 @@ otError otLinkRawSrcMatchEnable(otInstance *aInstance, bool aEnable);
  * @retval OT_ERROR_INVALID_STATE    If the raw link-layer isn't enabled.
  *
  */
-otError otLinkRawSrcMatchAddShortEntry(otInstance *aInstance, uint16_t aShortAddress);
+otError otLinkRawSrcMatchAddShortEntry(otInstance *aInstance, const uint16_t aShortAddress);
```

Comment:
Did you mean to add `const` here? Other similar instances below.

Reply:
Yep, the header didn't match the source file which caused linktime errors. I was a little confused why a `uint6_t` was being marked const, but I figured adding consts was better than removing them.

Reply:
Thanks for pointing that out. In this case, I think we should remove const. Indeed, we removed some of them in 71604f6cd0e19b6f3999ed949924c8691e922e17. However, after looking, it seems there are other instances. I'll submit a separate PR to clean this up more generally.

Reply:
@srickardti, just submitted #4901. Please review.